net/unison/Portfile

```diff
@@ -16,39 +15,41 @@ long_description    Unison is a file-synchronization tool for Unix and \
                     modified separately, and then brought up to date by \
                     propagating the changes in each replica to the other.
 homepage            http://www.cis.upenn.edu/~bcpierce/unison/
-master_sites        ${homepage}download/releases/${name}-${version}/
-platforms           darwin openbsd
+platforms           darwin
 
-set branch          [join [lrange [split ${version} .] 0 1] .]
+checksums           rmd160  24050d73c3d38389816828dd8ce77b4de7c80639 \
+                    sha256  8e071e862fef675ea034f0c3e4b881d392be0889db4f04ad894cc5a931891e3d \
+                    size    1400793
 
-checksums           rmd160  d5a1d66caffc02832c446f375a16319779fc5e0b \
-                    sha256  30aa53cd671d673580104f04be3cf81ac1e20a2e8baaf7274498739d59e99de8
+# from ocaml port
+supported_archs     i386 x86_64
+universal_variant   no
 
-patchfiles-append   patch-Makefile.diff
-platform darwin 8 {
-    patchfiles-append patch-Makefile.OCaml-MINOSXVERSION.diff
-}
-
-worksrcdir          src
+depends_build       port:ocaml
 
-use_configure       no
+installs_libs       no
 
-build.args          UISTYLE=text NATIVE=true THREADS=true
-depends_build       bin:camlp4:ocaml
+use_parallel_build  no
 
-destroot.args       PREFIX=${prefix} UISTYLE=text NATIVE=true THREADS=true
+patchfiles-append   patch-Makefile.OCaml.diff \
+                    patch-target.diff \
+                    patch-Makefile.diff \
+                    patch-check-cltool.diff
 
-default_variants    +symlink
+post-patch {
+    reinplace "s|__MACPORTS_PREFIX__|${prefix}|g" ${worksrcpath}/src/uimac14/MyController.m
 
-variant symlink description {Create symlink to versioned binary} {
-    post-destroot {
-        ln -s ${prefix}/bin/${name}-${branch} ${destroot}${prefix}/bin/${name}
-    }
 }
 
-variant x11 conflicts aqua {
-    patchfiles-append       patch-Makefile.OCaml.diff
+build.env-append    CC=${configure.cc}
+destroot.env-append PREFIX=${prefix}
+
+build.args          UISTYLE=text THREADS=true
+destroot.args       UISTYLE=text THREADS=true
```

Comment:
I note that you're not doing `NATIVE` any more. Was that deprecated? (I don't know Unison but I imagine that means generate a native executable and not OCamlrun bytecode. I would have to check to be sure.)

Reply:
I believe that ``NATIVE`` is ``true`` by default (see [here](https://github.com/bcpierce00/unison/blob/e1e047b038aed24f7d084e4dbb5b6a2c278e76c2/src/Makefile#L14)).

Reply:
Gotcha.